app/lib/io/zip.js

```diff
@@ -0,0 +1,34 @@
+define([
+    'jszip'
+  , 'io/readDirRecursive'
+], function(
+    JSZip
+  , readDirRecursive
+) {
+    "use strict";
+
+    var unpack = function(zipData, io, targetPath){
+        /* TODO: Implement-me! */
+    };
+
+    var encode = function(io, sourcePath){
+        var zip = new JSZip();
+        var files = readDirRecursive(false, io, sourcePath);
+        var i;
+
+        for (i in files){
+            var file = files[i];
+            var relative_path = file.split(sourcePath)[1];
+            var data = io.readFile(false, file);
+            zip.file(relative_path, data, {binary:true});
+        }
+
+        var content = zip.generate({type:"base64"});
```

Comment:
Is content in this case a `base64` string? What's the reasoning for this? `type` should probably be an argument of `encode`. Looking at: http://stuk.github.io/jszip/documentation/api_jszip/generate.html we have a problem to decide for a good option  between browser and nodeJS contexts.

The Ideal for browsers would be `blob` I think. It would be cool to have a compatibility strategy for nodeJS/Browser binary data objects. But I doubt that we'll have something sound soon.